lib/engine/game/g_18_carolinas.rb

```diff
@@ -7,6 +7,57 @@ module Engine
   module Game
     class G18Carolinas < Base
       load_from_json(Config::Game::G18Carolinas::JSON)
+
+      def init_companies(players)
+        super.take(players.size).each do |company|
+        end
+      end
+
+      def setup
+        counter = 0
+        @companies.each do |company|
+          price = company.value
```

Comment:
player = @players[index] # so you don't have to keep referencing the hash

Reply:
fixed, although I think I still need to reference it to give it the private and spend the money, right?